README.md

```diff
@@ -19,33 +19,296 @@ Terraform 0.12 and newer.
+| region | Region to work on. | string | |
+| env | The environment (e.g. prod, dev, stage) | string | "prod" |
+| clients | A list of client certificate name | list(string) | ["client"] |
+| cert_issuer | Common Name for CA Certificate | list(string) | "CA" |
+| cert_server_name | Name for the Server Certificate | string | "Server" |
+| aws_tenant_name | Name for the AWS Tenant | string | "AWS" |
+| key_save_folder | Where to store keys (relative to pki folder) | string | "clientvpn_keys" |
+| subnet_id | The subnet ID to which we need to associate the VPN Client Connection. | string | |
+| target_cidr_block | The CIDR block to wich the client will have access to. Might be VPC CIDR's block for example. | string | |
+| dns_servers | Information about the DNS servers to be used for DNS resolution. A Client VPN endpoint can have up to two DNS servers. | list(string) | null |
+| vpn_name | The name of the VPN Client Connection. | string | "VPN" |
+| cloudwatch_enabled | Indicates whether connection logging is enabled. | bool | true |
+| cloudwatch_log_group | The name of the cloudwatch log group. | string | vpn_endpoint_cloudwatch_log_group |
+| cloudwatch_log_stream | The name of the cloudwatch log stream. | string | vpn_endpoint_cloudwatch_log_stream |
+| aws_cli_profile_name | The name of the aws cli profile used in scripts | string | default |
+| client_auth | the type of client authentication to be used : certificate-authentication / directory-service-authentication / federated-authentication | string | certificate-authentication |
+| active_directory_id | The ID of the Active Directory to be used for authentication if type is directory-service-authentication | string | null |
+| root_certificate_chain_arn | The ARN of the client certificate. The certificate must be signed by a certificate authority (CA) and it must be provisioned in AWS Certificate Manager (ACM). Only necessary when type is set to certificate-authentication. | string | null |
+| saml_provider_arn | The ARN of the IAM SAML identity provider if type is federated-authentication | string | null |
+
+### Output
+| Name | Description |
+|------|-------------|
+| kms_sops_arn | |
+| decrypt_command | Output of the decrypt script |
+| encrypt_command | Output of the encrypt script |
+| server_certificate_arn | The ARN of the generated Server Certificate |
+| env | Environment variable |
+| pki_folder_name | Generated certificate folder |
+| client_vpn_endpoint_id | The ID of the Client VPN endpoint. |
+| client_vpn_endpoint_arn | The ARN of the Client VPN endpoint. |
+| client_vpn_endpoint_dns_name | The DNS name to be used by clients when establishing their VPN session. |
+| client_vpn_endpoint_status | The current state of the Client VPN endpoint. |
+
+## How does it work ?
+
+### 1. Server certificate generation
+Clone and use the latest easy-rsa repo to generate the ca and server certificate.
+All the PKI configuration files are saved into your terraform folder.
+```sh
```

Comment:
Might not be the best thing to add code directly into the README, maybe pseudocode would be better as the code might change but the overall logic is less likely to change.

Reply:
I agree with @Dr0p42 , if we keep it that way we would need to update the README at every code change

Reply:
I changed the readme